game/world/managers/objects/units/player/StatManager.py

```diff
@@ -415,6 +417,19 @@ def update_base_mana_regen(self):
         regen = CLASS_BASE_REGEN_MANA[player_class] + spirit * CLASS_SPIRIT_SCALING_MANA[player_class]
         self.base_stats[UnitStats.POWER_REGENERATION_PER_5] = int(regen / 2)
 
+    def update_base_melee_critical_chance(self):
+        if self.unit_mgr.get_type_id() != ObjectTypeIds.ID_PLAYER:
+            return
+
+        player_class = self.unit_mgr.player.class_
+        strength = self.get_total_stat(UnitStats.STRENGTH)
+        scaling = CLASS_STRENGTH_SCALING_CRITICAL[player_class]
+        class_rate = (scaling[0] * (60 - self.unit_mgr.level) +
+                      scaling[1] * (self.unit_mgr.level - 1)) / 59
+        crit = self.get_total_stat(UnitStats.CRITICAL)
+        total_crit = crit + strength / class_rate / 100
+        self.base_stats[UnitStats.CRITICAL] = total_crit
```

Comment:
You shouldn't use `get_total_stat(UnitStats.CRITICAL)` here. It's used for applying all bonuses to a stat, but you should only be calculating the stat gain from base stats. You can for example check `update_base_dodge_chance` since the logic should be almost the same.